.github/workflows/main.yml

```diff
@@ -0,0 +1,21 @@
+on: [push]
+
+jobs:
+  lint_and_test:
+    runs-on: ubuntu-latest
+    name: run ci lint + tests
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      # Is this really the best way to do this?
+      # I am not happy at all with how I've gotten docker and compose
+      # configured, but not sure what the best way to do it is :/
+      #
+      # Ideally we have some other process that builds the dockerfile.
+      - name: build the stack
+        run: docker compose up -d
```

Comment:
https://docs.github.com/en/actions/use-cases-and-examples/using-containerized-services/about-service-containers?

Reply:
promising, but I've found some resources around using docker compose within Github. I kinda don't like the idea of having to declare the service dependencies in multiple places? IDK - I just feel like I should probably commit to using docker compose so local runs are consistent with stuff in the cloud always. 🫨 🤷